docs/architecture/engine.md

```diff
@@ -0,0 +1,270 @@
+# Storage Engine Overview
+
+M3DB is a time series Database that was primarily designed to be horizontally scalable and handle a large volume of monitoring time series data.
+
+## Caveats / Limitations
+
+M3DB currently does not support the following:
+
+1. Any form of indexing. While M3DB is useful as a horizontally scalable datastore that provides extremely high compression ratios for time series data, it does not currently perform any type of indexing. This feature is currently under development and future versions of M3DB will have support for a built-in index that can be used standalone for smaller M3DB clusters (up to 10 nodes).
+2. Updates / deletes. All data written to M3DB is immutable.
+3. Writing very far into the past and future. M3DB was originally designed for storing high volumes of monitoring data, and thus writing data at arbitrary timestamps in the past and future was never an intended design goal. That said, this feature is currently under development and future versions of M3DB will have support for this.
+4. Writing datapoints with values other than double-precision floats. We have planned work on our roadmap to improve the state of affairs so that arbitrary data types can be encoded.
+5. Storing data permanently. Every namespace in M3DB is required to have a retention policy which specifies how long data in that namespace will be retained for. While there is no upper bound on that value (Uber has production databases running with retention periods as high as 5 years), its still required and generally speaking M3DB is optimized for workloads with a well-defined [TTL](https://en.wikipedia.org/wiki/Time_to_live).
+6. Writing / reading from languages other than Golang. Communicating with M3DB currently requires use of a "fat" (logic-heavy) client, and there is only a Golang implementation at the moment.
```

Comment:
Not exactly true. We do support JSON over HTTP for the `cluster` endpoints. NB: these aren't the `node` performant endpoints afair

Reply:
Any suggestions on how to phrase that? My understanding is that you would have to implement consistency at the application layer if you went around the Go client